Fixed Issue: Special price of 0.0000 is not shown on frontend, but is calculated in cart

mahesh-rajawat opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

Fixed issue of product with a special price of 0.00, this is not shown on the frontend. Not in the product listings and also not on the product detail page.

### Description (*)
Product Special Price is set to 0 then its not visible on product listing and Product View page, but when
we add product in cart, its price added to 0 in cart. In this pull request i have fixed this issue, now
Special price is displaying 0 as well. 

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/magento2#18268: Special price of 0.0000 is not shown on frontend, but is calculated in car

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Created new product and set the special price 0 and saved it.
2. Open product in category page special price is displaying 0 as normal.
3. Open product in product view page special price is displaying 0.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

AnshuMishra17 committed:
Refactor Mass Order Cancel code to use Interface

I have observed that MassAction Cancel is using the collection for orders cancel, whereas Order Cancel (order cancel from order edit section) is using Interface to put the order on hold.
So, I have refactor the Mass Order Cancel code to use Sales Order Interface.
# Conflicts:
#	app/code/Magento/Sales/Controller/Adminhtml/Order/MassCancel.php

magento-engcom-team committed:
ENGCOM-1266: Refactor Mass Order Cancel code to use Interface #14630

- Merge Pull Request magento/magento2#14630 from AnshuMishra17/magento2:patch-2
 - Merged commits:
   1. dab8c3897444df70498b2388735b2d06e218d5f3
   2. 97f5978c0e8c7b3d042ce74ca44eca691729b860
   3. 2b4ca2e0205a6b4edf15f36c6e5e983f0fdeb981

# Conflicts:
#	app/code/Magento/Sales/Controller/Adminhtml/Order/MassCancel.php
#	app/code/Magento/Sales/Test/Unit/Controller/Adminhtml/Order/MassCancelTest.php

Commit:
Improve code quality for Subscriber/NewAction

+ Refactor `else` block for returning success message to private function
+ Use import statements for classes
+ Because imports are being used an `if` statement can be written on one line
+ Use `addExceptionMessage` and `addSuccessMessage` instead of their deprecated counterparts
+ Include with DI the `Magento\Framework\Validator\EmailAddress` class
which can be used to validate the email address. Therefore the static
function call to the `is` method on the `Zend_Validate` class is not
needed anymore.

mage2pratik committed:
Removed commented code

mage2pratik committed:
Removed commented code from lib files

Floctopus committed:
Fix documentation typos in registry.js

Fixed multiple typos in php docblocks.

deefco committed:
Fix typos and grammar errors in actions.js

EmilyPepperman committed:
small misspelling fixed

Commit:
Fix wrong return type in StockRegistryInterface API

mage2pratik committed:
Removed commented code from lib files

jignesh-baldha committed:
Removed unnecessary characters from comments

jignesh-baldha committed:
Removed unnecessary characters from comments #2

vgoncharenko committed:
MAGETWO-94384: Apply profile changes

rogyar committed:
Added integration tests for gift message quote merge

hryvinskyi committed:
Fixing Snake Case To Camel Case

ishakhsuvarov committed:
magento/magento2#18064: Fixing Snake Case To Camel Case - static test fix

dmytro-ch committed:
Sales: add missing unit tests for model classes

Commit:
Fix table rate failing for zip+4 address #17770

Commit:
Fix for custom product attribute changing 'backend_type' when 'is_user_defined = 1' and get updated/saved in Admin Backend

Fix for custom product attribute changing 'backend_type' when 'is_user_defined = 1' and get updated/saved in Admin Backend

Fix for custom product attribute changing 'backend_type' when 'is_user_defined = 1' and get updated/saved in Admin Backend

hostep committed:
Backport 2.2 - Introducing a dedicated cron.log file for logging cron related info, this reduces output in the system.log file.

(cherry picked from commit a4f58c8806d93ff20f921e02785a78bda00adf46)

hostep committed:
Backport 2.2 - Allow keyboard navigation in browser on product detail pages, by fixing fotorama library to not change to a different image when pressing left or right arrow key in combination with certain modifier keys on your keyboard.

(cherry picked from commit d134a232893490dc4e0d7e520d15eef9b36f23e0)

VincentMarmiesse committed:
Label should always be blank even if attribute is required

hostep committed:
Backport 2.2 - Don't set a source model on the attribute when it's not needed. This avoids accidentally persisting the source model to the database when using multiselect attributes.

(cherry picked from commit 477ca51d052464e862301d945c8bb3550b6fa829)

magento-engcom-team committed:
ENGCOM-3052: [Backport] Change sort order for customer group options #18280

 - Merge Pull Request magento/magento2#18280 from dmytro-ch/magento2:2.2-develop-PR-port-18117
 - Merged commits:
   1. 484bbba6407f03c7b2261992c0ec2433d00a2cb4
   2. dd9b2948a03abdc487e712d4237a9180484cb17b

Commit:
Fixed code style issue

magento-engcom-team committed:
ENGCOM-3054: fix wysiwyg editor not decoding base64 filenames special chars #18215

 - Merge Pull Request magento/magento2#18215 from adammada/magento2:wysiwyg-fix
 - Merged commits:
   1. 4e62fe32bf6f27b583eeb8d5d2309c7abebd5535
   2. bf2efffff0da82f33796c82d643403545466911d

viktym committed:
Merge pull request #3219 from magento-tsg-csl3/2.2-develop-pr3

[TSG-CSL3] Backporting 2.2 (pr3)

haroldclaus committed:
Fix for removing the dirs while creating a TAR archive

The bug caused issues during the export of analytics data from advanced
reporting. The scandir has parameter were the sort can be placed, but
using UNSORTED gives you unwanted lists of files/dirs. Therefor deleting
the first 2 via array_shift, is not a proper way of dealing with this

Signed-off-by: Harold Claus <harold.claus@studioemma.com>

svitja committed:
MAGETWO-91105: Table Rates not working for Armed Forces states in Europe (AE)

magento-engcom-team committed:
ENGCOM-3047: Cast products 'getStoreId()' to int, closes #18079 #18086

 - Merge Pull Request magento/magento2#18086 from sreichel/magento2:develop/fix-getStoreId
 - Merged commits:
   1. 9d8b04a0c9700e8a5bfe65911c8151672f202de6

zakdma committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-73526

Commit:
ENGCOM-3054: fix wysiwyg editor not decoding base64 filenames special chars #18215

Commit:
ENGCOM-3052: [Backport] Change sort order for customer group options #18280

Commit:
ENGCOM-3047: Cast products "getStoreId()" to int, closes #18079 #18086

Commit:
ENGCOM-2988: [Fix] Do not modify current list of countries with require states during setup upgrade #16885

magento-engcom-team committed:
ENGCOM-3057: Fix table rate failing for zip+4 address #17770 #18166

 - Merge Pull Request magento/magento2#18166 from magently/magento2:fix/extended-zip-code-in-table-rates
 - Merged commits:
   1. 6e4a4b2d85e25c58e72914a7904a5dc9aa7e43f5

okolesnyk committed:
Merge branch '2.2-develop' into 2.2-dev-pan-2.3.0

yelahin-serhiy committed:
MAGETWO-94154: Files and folders symlinked in pub/media cannot be deleted from media gallery browser

svitja committed:
MAGETWO-86303: Updated url keys of product after import produce 404 error

svitja committed:
MAGETWO-91105: Table Rates not working for Armed Forces states in Europe (AE)

OlgaVasyltsun committed:
MAGETWO-89296: Menu in mobile view cannot be closed

svitja committed:
MAGETWO-91105: Table Rates not working for Armed Forces states in Europe (AE)

Commit:
MAGETWO-94295: United States 'state' field in admin orders is not a drop down

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#18215: fix wysiwyg editor not decoding base64 filenames special chars (by @adammada)
 - magento/magento2#18280: [Backport] Change sort order for customer group options (by @dmytro-ch)
 - magento/magento2#18086: Cast products "getStoreId()" to int, closes #18079 (by @sreichel)
 - magento/magento2#16885: [Fix] Do not modify current list of countries with require states during setup upgrade (by @jalogut)


Fixed GitHub Issues:
 - magento/magento2#18138: WYSIWYG editor fails to parse directives of files with special characters in URL (so random files) (reported by @adammada) has been fixed in magento/magento2#18215 by @adammada in 2.2-develop branch
   Related commits:
     1. 4e62fe32bf6f27b583eeb8d5d2309c7abebd5535
     2. bf2efffff0da82f33796c82d643403545466911d

 - magento/magento2#18101: Wrong sort order for customer groups in customer grid filter (reported by @sreichel) has been fixed in magento/magento2#18280 by @dmytro-ch in 2.2-develop branch
   Related commits:
     1. 484bbba6407f03c7b2261992c0ec2433d00a2cb4
     2. dd9b2948a03abdc487e712d4237a9180484cb17b

 - magento/magento2#18079: Inconsistent return type for getStoreId() (reported by @sreichel) has been fixed in magento/magento2#18086 by @sreichel in 2.2-develop branch
   Related commits:
     1. 9d8b04a0c9700e8a5bfe65911c8151672f202de6

vgoncharenko committed:
MAGETWO-94384: Apply profile changes

svitja committed:
MAGETWO-86303: Updated url keys of product after import produce 404 error

Commit:
Merge branch 'MAGETWO-94181' into mpi-PR-2809

duhon committed:
MAGETWO-95075: Fix cache hit ratio on measurements

Commit:
merge magento/2.2-develop into magento-pangolin/2.2-dev-pan-2.3.0

duhon committed:
Merge remote-tracking branches 'helix/MAGETWO-95075' and 'helix/MAGETWO-94384' into pr_helix_2.2

okolesnyk committed:
MQE-1112: Bump MFTF version in Magento

okolesnyk committed:
Merge branch '2.2-dev-pan-2.3.0' of github.com:magento-pangolin/magento2ce into 2.2-dev-pan-2.3.0

duhon committed:
Merge pull request #3229 from magento-helix/pr_helix_2.2

[helix] MAGETWO-93293: B2C measurements for 2.2.6&2.2.5
- MAGETWO-95075 Fix cache hit ratio on measurements

aljcalandra committed:
MQE-1112: Bump MFTF version in Magento

- Removed `before` specified for two actions in verifyBasicOrderInformation actionGroup

dmytro-ch committed:
Fix excessively long variable names

rogyar committed:
Minor code style fix

magento-engcom-team committed:
ENGCOM-3065: [Backport] Sales: add missing unit tests for model classes #18310

 - Merge Pull Request magento/magento2#18310 from dmytro-ch/magento2:2.2-develop-PR-port-18072
 - Merged commits:
   1. 231f09708bd219715bdc22bc81cf5c95f68328e0
   2. c65b5a412670900b515f018b368987d902e08d3f
   3. 828e742371da1c42fa1e4b2bc07c876b410df099

magento-engcom-team committed:
ENGCOM-3066: [Backport] Added integration test for gift message quote merge #18311

 - Merge Pull Request magento/magento2#18311 from dmytro-ch/magento2:2.2-develop-PR-port-18047
 - Merged commits:
   1. 82c4698b4a48afac70f9dadd6631c1bba10925f5

ihor-sviziev committed:
Fix static tests

svitja committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-73526

Commit:
Merge branch '2.2-develop' of github.com:magento/magento2ce into mpi-PR-2809

Commit:
merge magento/2.2-develop into magento-pangolin/2.2-dev-pan-2.3.0

viktym committed:
Merge pull request #3234 from magento-mpi/mpi-PR-2809

[MPI]-MAGETWO-94181

ihor-sviziev committed:
Fix throwing error by checkout error processor model

svitja committed:
MAGETWO-91105: Table Rates not working for Armed Forces states in Europe (AE)

Commit:
merge magento/2.2-develop into magento-pangolin/2.2-dev-pan-2.3.0

jilu1 committed:
Merge remote-tracking branch 'origin/2.2-develop' into MQE-1139-2.2

OlgaVasyltsun committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-89296

svitja committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-91105

svitja committed:
MAGETWO-73332: Update Catalog Rule Test

rostyslav-hymon committed:
MAGETWO-93818: Magnifier function does not disappear after mouse-off the image from the bottom

svitja committed:
MAGETWO-91105: Table Rates not working for Armed Forces states in Europe (AE)

Commit:
MAGETWO-93262: Cart is not empty after removing the product

Commit:
ENGCOM-3066: [Backport] Added integration test for gift message quote merge #18311

Commit:
ENGCOM-3065: [Backport] Sales: add missing unit tests for model classes #18310

Commit:
ENGCOM-3044: Fixed issue with lib-line-height mixin failing when value of 'normal'… #18168

rogyar committed:
Added unit test for CRON converter plugin

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#18311: [Backport] Added integration test for gift message quote merge (by @dmytro-ch)
 - magento/magento2#18310: [Backport] Sales: add missing unit tests for model classes (by @dmytro-ch)
 - magento/magento2#18168: Fixed issue with lib-line-height mixin failing when value of 'normal' (by @CNanninga)

slavvka committed:
Fixed codestyle

magento-engcom-team committed:
ENGCOM-3079: add error message in else condition #17982

 - Merge Pull Request magento/magento2#17982 from vaibhavahalpara/magento2:issue-17977
 - Merged commits:
   1. 3d13f783c9b1e8b9d48bb0b23819598c3af65d17
   2. 7107b8da49c06e783b7629ed3b88757bd97f53e1
   3. 49f6549c682a7fb807faee247ad8f805a82d0f12

yelahin-serhiy committed:
Merge remote-tracking branch 'origin/MAGETWO-94462' into 2.2-develop-pr4

yelahin-serhiy committed:
Merge remote-tracking branch 'origin/MAGETWO-94154' into 2.2-develop-pr4

rostyslav-hymon committed:
Merge branch 'MAGETWO-93818' into 2.2-develop-pr4

Commit:
Merge branch 'MAGETWO-93262' into 2.2-develop-pr4

Commit:
Merge branch 'MAGETWO-94295' into 2.2-develop-pr4

Commit:
Fix for parsing attribute options labels, when & used.

svitja committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-91105

svitja committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-73332

magento-engcom-team committed:
ENGCOM-3081: Fix for parsing attribute options labels, when & used. #18354

 - Merge Pull Request magento/magento2#18354 from bartoszkubicki/magento2:preserve-&-in-attribute-option-label
 - Merged commits:
   1. e6dc18822c80d7df4e7ebfe6cb4a93c0165624c0

Thundar committed:
Issue 12479 refactor, attribute_set_id always not null on customer and customer address

Commit:
ENGCOM-3079: add error message in else condition #17982

Commit:
ENGCOM-2832: ConfigurableProduct show prices in select options #17695

ameysar committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-73332

viktym committed:
Merge pull request #3243 from magento-tsg-csl3/2.2-develop-pr4

[TSG-CSL3] Backporting 2.2 (pr4)

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

Commit:
ENGCOM-3081: Fix for parsing attribute options labels, when & used. #18354

Commit:
merge magento/2.2-develop into magento-pangolin/2.2-dev-pan-2.3.0

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#17982: add error message in else condition (by @vaibhavahalpara)
 - magento/magento2#17695: ConfigurableProduct show prices in select options (by @alexeya-ven)
 - magento/magento2#18354: Fix for parsing attribute options labels, when & used. (by @bartoszkubicki)


Fixed GitHub Issues:
 - magento/magento2#17977: Show Method if Not Applicable for Free Shipping doesn't work. (reported by @vaibhavahalpara) has been fixed in magento/magento2#17982 by @vaibhavahalpara in 2.2-develop branch
   Related commits:
     1. 3d13f783c9b1e8b9d48bb0b23819598c3af65d17
     2. 7107b8da49c06e783b7629ed3b88757bd97f53e1
     3. 49f6549c682a7fb807faee247ad8f805a82d0f12

Commit:
merge magento/2.2-develop into magento-pangolin/2.2-dev-pan-2.3.0

magento-engcom-team committed:
ENGCOM-2916: Do not overwrite URL Key with blank value #17882

 - Merge Pull Request magento/magento2#17882 from josephmcdermott/magento2:patch-1
 - Merged commits:
   1. a021d5f72b2c23be2820225d869b72a48d437530
   2. a02bf03e26c3409a0813e4dc4e74aa8ccf21eb51
   3. e48683cd6827ff527a7f22f6c2e39a8a2922bd9a
   4. 7df4e129dd40e1659e4f81554d1a8ba37e6850f8
   5. fffb6ff6e8b4302db2f1b824fbe96a5cdacd90ff
   6. 002aa19711634a906aab53d6e1110babf604892e
   7. 172e2ec1ffd24e04e0050e58b7cb090785ebd9a1
   8. 3a54946f843d469b441185a36fc1a986e7e941fc
   9. 2fb71834f2d0ce7282afda3d350cf404ad35be8f
   10. fda664a268303561cd3acfbb35b52f85ae525ce5
   11. abd95bdc0ab572881f8b341902d7115900fd0e39
   12. 0f26b52735389f412eb883eec978cd12052e0d97

magento-engcom-team committed:
ENGCOM-3087: [Backport] Fix for removing the dirs while creating a TAR archive #18283

 - Merge Pull Request magento/magento2#18283 from haroldclaus/magento2:hotfix/archiveTarRemoveDirsFix
 - Merged commits:
   1. 45f22aca2e455f2df3fdb0f7ae91563cf388e407
   2. e1d2673086323bc3d633e822afefb5e4d3a22071

magento-engcom-team committed:
ENGCOM-3088: [Backport] Fix throwing error by checkout error processor model #18369

 - Merge Pull Request magento/magento2#18369 from ihor-sviziev/magento2:2.2-develop-PR-port-18331
 - Merged commits:
   1. 74bed695b4ac0f381b10368c7d8a780b52b5c9f9

svitja committed:
MAGETWO-73332: Update Catalog Rule Test

svitja committed:
Merge remote-tracking branch 'origin/MAGETWO-73332' into MAGETWO-73332

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-73526' into 2.2-develop-pr48

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-86686' into 2.2-develop-pr48

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-94179' into 2.2-develop-pr48

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-89986' into 2.2-develop-pr48

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-94262' into 2.2-develop-pr48

sambolek committed:
Backport 2.2 - Fix wrong reference in google analytics module layout xml

magento-engcom-team committed:
ENGCOM-3091: Backport 2.2 - Fix wrong reference in google analytics module layout xml #18375

 - Merge Pull Request magento/magento2#18375 from woomdigital/magento2:fix-issue-16497-backport-22
 - Merged commits:
   1. 018901480f9d9be3e7813d7b91d32cbd45f82140

sambolek committed:
Backport 2.2 - Fix issue 17152 - prevent email being marked as not sent if email copy fails due to exception

magento-engcom-team committed:
ENGCOM-3092: Backport 2.2 - Fix issue 17152 - prevent email being marked as not se… #18376

 - Merge Pull Request magento/magento2#18376 from woomdigital/magento2:fix-issue-17152-backport-22
 - Merged commits:
   1. 058c254f0d39a6ab1cad94de42863117404cdfb5

Commit:
Remove duplicate

Commit:
Fixed code style issue

magento-engcom-team committed:
ENGCOM-3093: Implemented 17964: Backend Order creation Authorizenet: If invalid cr… #17986

 - Merge Pull Request magento/magento2#17986 from passtet/magento2:github-17964
 - Merged commits:
   1. 41def56640e8f91369145ce81632dc0a60565c27
   2. 1f08cd108688d742a5d66a944f142c5738b81ec0
   3. edea5c57706895a3384e35a589fcb2352640b509

Commit:
Removed double call of orderManagement service

magento-engcom-team committed:
ENGCOM-3094: [Backport 2.2-develop] Refactor Mass Order Cancel code to use Interface #18377

 - Merge Pull Request magento/magento2#18377 from JeroenVanLeusden/magento2:patch-1
 - Merged commits:
   1. e803e6a7f23d9dd5a5aa1c33b9235097c5058052
   2. 21dfeaeec760044547c1b7e10f3b0b713ecc6d2c
   3. 0ecf1fae15f0e481d21bc0114ae8d35195d56038
   4. 2569a0bf546008c9734155a928a968ee1e954b63

nmalevanec committed:
ENGCOM-2997: #4942 and bundle checkbox bug #15905. Fix functional and static tests.

Commit:
Fix Failed MassCancelTest

ameysar committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-88254

nmalevanec committed:
Merge branch '2.2-develop' into ENGCOM-2104-magento-magento2-16115

Commit:
ENGCOM-3091: Backport 2.2 - Fix wrong reference in google analytics module layout xml #18375

Commit:
ENGCOM-3088: [Backport] Fix throwing error by checkout error processor model #18369

Commit:
ENGCOM-3087: [Backport] Fix for removing the dirs while creating a TAR archive #18283

Commit:
ENGCOM-3094: [Backport 2.2-develop] Refactor Mass Order Cancel code to use Interface #18377

Commit:
ENGCOM-3093: Implemented 17964: Backend Order creation Authorizenet: If invalid cr… #17986

Commit:
ENGCOM-2916: Do not overwrite URL Key with blank value #17882

Commit:
Merge pull request #3254 from magento-tsg/2.2-develop-pr48

[TSG] Backporting for 2.2 (pr48) (2.2.8)

Commit:
merge magento/2.2-develop into magento-pangolin/2.2-dev-pan-2.3.0

nmalevanec committed:
Merge branch '2.2-develop' into ENGCOM-2104-magento-magento2-16115

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#18375: Backport 2.2 - Fix wrong reference in google analytics module layout xml (by @sambolek)
 - magento/magento2#18369: [Backport] Fix throwing error by checkout error processor model (by @ihor-sviziev)
 - magento/magento2#18283: [Backport] Fix for removing the dirs while creating a TAR archive (by @haroldclaus)
 - magento/magento2#18377: [Backport 2.2-develop] Refactor Mass Order Cancel code to use Interface (by @JeroenVanLeusden)
 - magento/magento2#17986: Implemented 17964: Backend Order creation Authorizenet: If invalid cr (by @passtet)
 - magento/magento2#17882: Do not overwrite URL Key with blank value (by @josephmcdermott)


Fixed GitHub Issues:
 - magento/magento2#16497: Magento 2.2.5: Google Analytics not added to head correctly (reported by @WalterSmulders) has been fixed in magento/magento2#18375 by @sambolek in 2.2-develop branch
   Related commits:
     1. 018901480f9d9be3e7813d7b91d32cbd45f82140

 - magento/magento2#18330: Checkout - Infinite loading indicator when server returned error (reported by @ihor-sviziev) has been fixed in magento/magento2#18369 by @ihor-sviziev in 2.2-develop branch
   Related commits:
     1. 74bed695b4ac0f381b10368c7d8a780b52b5c9f9

 - magento/magento2#17023: CSV Import of `sku,attribute` empties `url_key` value (reported by @josephmcdermott) has been fixed in magento/magento2#17882 by @josephmcdermott in 2.2-develop branch
   Related commits:
     1. a021d5f72b2c23be2820225d869b72a48d437530
     2. a02bf03e26c3409a0813e4dc4e74aa8ccf21eb51
     3. e48683cd6827ff527a7f22f6c2e39a8a2922bd9a
     4. 7df4e129dd40e1659e4f81554d1a8ba37e6850f8
     5. fffb6ff6e8b4302db2f1b824fbe96a5cdacd90ff
     6. 002aa19711634a906aab53d6e1110babf604892e
     7. 172e2ec1ffd24e04e0050e58b7cb090785ebd9a1
     8. 3a54946f843d469b441185a36fc1a986e7e941fc
     9. 2fb71834f2d0ce7282afda3d350cf404ad35be8f
     10. fda664a268303561cd3acfbb35b52f85ae525ce5
     11. abd95bdc0ab572881f8b341902d7115900fd0e39
     12. 0f26b52735389f412eb883eec978cd12052e0d97

magento-engcom-team committed:
ENGCOM-3097: [FEATURE] [issue-3283] Added Filter Support for Yes/No (boolean) attr… #17823

 - Merge Pull Request magento/magento2#17823 from experius/magento2:experius-2.2-patch-issue-3283-boolean-filterable
 - Merged commits:
   1. 952b333b9908121f17e966078737c21b8620cce6
   2. a2d9a6a5f84d680c23d4f269596baf40ac709395

Commit:
merge magento/2.2-develop into magento-pangolin/2.2-dev-pan-2.3.0

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-73332' into 2.2-develop-pr49

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-86303' into 2.2-develop-pr49

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-88254' into 2.2-develop-pr49

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-95059' into 2.2-develop-pr49

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-89296' into 2.2-develop-pr49

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-91105' into 2.2-develop-pr49

magento-engcom-team committed:
ENGCOM-3101: Backport 2.2 - Allow keyboard navigation in browser on product detail… #18391

 - Merge Pull Request magento/magento2#18391 from hostep/magento2:allow-browser-keyboard-navigation-on-product-detail-pages-for-2.2
 - Merged commits:
   1. 518b0bb91c9b55cfd6df240271d2151d041368f2

ameysar committed:
MAGETWO-88254: [Magento Cloud] "Tier Pricing" of Products changes to "Price" (without discount) after Updated Items and Quantities in the Order.

ameysar committed:
Merge remote-tracking branch 'origin/MAGETWO-88254' into 2.2-develop-pr49

Commit:
Merge pull request #3259 from magento-tsg/2.2-develop-pr49

[TSG] Backporting for 2.2 (pr49) (2.2.8)

rafaelstz committed:
Admin Login Form > Aliging Label

Commit:
merge magento/2.2-develop into magento-pangolin/2.2-dev-pan-2.3.0

rafaelstz committed:
Admin > Footer > Aligning Proportionally

rafaelstz committed:
Setup Wizard > Menu Logo Bugfix

magento-engcom-team committed:
ENGCOM-3104: Backport 2.2 - Introducing a dedicated cron.log file for logging cron… #18389

 - Merge Pull Request magento/magento2#18389 from hostep/magento2:fix-for-issue-17190-for-2.2
 - Merged commits:
   1. a251a8d36ec2c10d175302cac3675e97990bce47

okolesnyk committed:
MQE-1112: Bump MFTF version in Magento

okolesnyk committed:
MQE-1112: Bump MFTF version in Magento

magento-engcom-team committed:
ENGCOM-3105: Admin Login Form > Aliging Label #18400

 - Merge Pull Request magento/magento2#18400 from rafaelstz/magento2:login-form-padding
 - Merged commits:
   1. af700494b83600a29d03badbef20374a81960b4a

dmytro-ch committed:
Fix the issue with customer inline edit when password is expired, updated the unit tests #18162

dmytro-ch committed:
Fix the issue with incorrect imports

magento-engcom-team committed:
ENGCOM-3109: [Backport] Added unit test for CRON converter plugin #18415

 - Merge Pull Request magento/magento2#18415 from dmytro-ch/magento2:2.2-develop-PR-port-18344
 - Merged commits:
   1. 01d374f80244afe1fd53b8680e075328136b1617

ArtsiomBruneuski committed:
throw exception InvalidArgumentException during validate scheme

ArtsiomBruneuski committed:
catch InvalidArgumentException, throw correct message, unit test

ArtsiomBruneuski committed:
annotations

ArtsiomBruneuski committed:
remove annotations

ArtsiomBruneuski committed:
throw exception code from previous exception

ArtsiomBruneuski committed:
modified exception message

ArtsiomBruneuski committed:
Was added description to exceptions, annotations to properties and classes. Implemented validation of exception message in unit test

lewisvoncken committed:
[BACKPORT] Replace sort callbacks to spaceship operator

magento-engcom-team committed:
ENGCOM-3111: [BACKPORT] Replace sort callbacks to spaceship operator #18422

 - Merge Pull Request magento/magento2#18422 from experius/magento2:experius-2.2-patch-backport-18187
 - Merged commits:
   1. a59736992e8d8e022158c17ffb8909f0117845d3

magento-engcom-team committed:
ENGCOM-3112: [Backport] small misspelling fixed #18428

 - Merge Pull Request magento/magento2#18428 from lewisvoncken/magento2:2.2-develop-PR-port-17470
 - Merged commits:
   1. ac3e0a767c930a64a01e5d28847cbaffd5855719

magento-engcom-team committed:
ENGCOM-3113: [Backport] Removed unnecessary characters from comments #18426

 - Merge Pull Request magento/magento2#18426 from lewisvoncken/magento2:2.2-develop-PR-port-17814
 - Merged commits:
   1. a0b0b3bcf17969e4f96b2281e9b3bb72ef9af0ca
   2. a48bb8125effa8cc8201415aec0fa1035c12d07d

magento-engcom-team committed:
ENGCOM-3114: [Backport] Fixing Snake Case To Camel Case #18425

 - Merge Pull Request magento/magento2#18425 from lewisvoncken/magento2:2.2-develop-PR-port-18064
 - Merged commits:
   1. fa99cd6232116d27024573efa9776a69890a2a7a
   2. bc775a5872e33165ecd5818231b4097abe00baeb

ArtsiomBruneuski committed:
short descriptions was added

magento-engcom-team committed:
ENGCOM-3115: [Backport] Fix documentation grammar errors and typos in actions.js #18429

 - Merge Pull Request magento/magento2#18429 from lewisvoncken/magento2:2.2-develop-PR-port-17462
 - Merged commits:
   1. f981d23bc8bc0b092bdd95137505b6b2c07d9fa0

magento-engcom-team committed:
ENGCOM-3107: [Backport] Fix the issue with customer inline edit when password is expired #18414

 - Merge Pull Request magento/magento2#18414 from dmytro-ch/magento2:fix/2.2-issue-18162
 - Merged commits:
   1. 33ba243a068b9e695fdb2520ab463aa3d5911f1a
   2. 2668f2df768d749f2069195715188afbf6d99348

ArtsiomBruneuski committed:
moved strict_types block

magento-engcom-team committed:
ENGCOM-3120: [Backport] Improve code quality subscriber new action #18433

 - Merge Pull Request magento/magento2#18433 from lewisvoncken/magento2:2.2-develop-PR-port-17232
 - Merged commits:
   1. 8e226765106d34aba3cddd68c90aa1bd3c508030

magento-engcom-team committed:
ENGCOM-3121: [Backport] Removed commented code #18432

 - Merge Pull Request magento/magento2#18432 from lewisvoncken/magento2:2.2-develop-PR-port-17346
 - Merged commits:
   1. 5fbef2bda9371b232d709d5053c59109989d3057
   2. 5c0496803499e5000df20c2f260ab3751148406b
   3. 2ac34d30ba30887b5d0e3fecdceb92535feda3d4

magento-engcom-team committed:
ENGCOM-3123: [Backport] Fix documentation typos in registry.js #18430

 - Merge Pull Request magento/magento2#18430 from lewisvoncken/magento2:2.2-develop-PR-port-17430
 - Merged commits:
   1. 1c9510f5bdf4a8a3a28334987f1437e3ab9f5ff9

ArtsiomBruneuski committed:
removed wrong symbol

okolesnyk committed:
Merge pull request #3159 from magento-pangolin/2.2-dev-pan-2.3.0

[pangolin] MFTF 2.3.0 version update for CE 2.2

magento-engcom-team committed:
ENGCOM-3126: Fix setup wizard page logo #18403

 - Merge Pull Request magento/magento2#18403 from rafaelstz/magento2:setup-page-logo
 - Merged commits:
   1. 6b3a59b242bb37980aa2bb8eb25c1119bfb8d577

magento-engcom-team committed:
ENGCOM-3128: Backport 2.2 - Don't set a source model on the attribute when it's no… #18390

 - Merge Pull Request magento/magento2#18390 from hostep/magento2:fix-for-issue-13156-for-2.2
 - Merged commits:
   1. 3e066db75e495f96d264bf1347b9fec01a44032e

nmalevanec committed:
ENGCOM-2629: Added checks to see if the payment is available #15683

magento-engcom-team committed:
ENGCOM-3133: [Backport] Fix wrong return type in StockRegistryInterface API #18427

 - Merge Pull Request magento/magento2#18427 from lewisvoncken/magento2:2.2-develop-PR-port-17562
 - Merged commits:
   1. adfb130b1bc7074daaeaed466f07a283e2eee8ca

Commit:
ENGCOM-3123: [Backport] Fix documentation typos in registry.js #18430

Commit:
ENGCOM-3121: [Backport] Removed commented code #18432

Commit:
ENGCOM-3120: [Backport] Improve code quality subscriber new action #18433

Commit:
ENGCOM-3115: [Backport] Fix documentation grammar errors and typos in actions.js #18429

Commit:
ENGCOM-3113: [Backport] Removed unnecessary characters from comments #18426

Commit:
ENGCOM-3112: [Backport] small misspelling fixed #18428

Commit:
ENGCOM-3109: [Backport] Added unit test for CRON converter plugin #18415

Commit:
ENGCOM-3107: [Backport] Fix the issue with customer inline edit when password is expired #18414

Commit:
ENGCOM-3105: Admin Login Form > Aliging Label #18400

Commit:
ENGCOM-3101: Backport 2.2 - Allow keyboard navigation in browser on product detail… #18391

Commit:
ENGCOM-3092: Backport 2.2 - Fix issue 17152 - prevent email being marked as not se… #18376

jilu1 committed:
Merge remote-tracking branch 'origin/2.2-develop' into MQE-1139-2.2

# Conflicts:
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/SampleTests/Test/AdvancedSampleTest.xml
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/SampleTests/Test/AssertsTest.xml
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/SampleTests/Test/CreateConfigurableProductByApiTest.xml
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/SampleTests/Test/CreateSalesRuleByApiTest.xml
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/SampleTests/Test/PersistMultipleEntitiesTest.xml
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/SampleTests/Test/SetPaymentConfigurationTest.xml
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/SampleTests/Test/UpdateSimpleProductByApiTest.xml

jilu1 committed:
MQE-1139: cleanup test modules in dev/tests/acceptance

- moved some newly added tests to correct magento module

Commit:
magento-engcom/magento2ce#2227 Fixed test failures

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#18430: [Backport] Fix documentation typos in registry.js (by @lewisvoncken)
 - magento/magento2#18432: [Backport] Removed commented code (by @lewisvoncken)
 - magento/magento2#18433: [Backport] Improve code quality subscriber new action (by @lewisvoncken)
 - magento/magento2#18429: [Backport] Fix documentation grammar errors and typos in actions.js (by @lewisvoncken)
 - magento/magento2#18426: [Backport] Removed unnecessary characters from comments (by @lewisvoncken)
 - magento/magento2#18428: [Backport] small misspelling fixed (by @lewisvoncken)
 - magento/magento2#18415: [Backport] Added unit test for CRON converter plugin (by @dmytro-ch)
 - magento/magento2#18414: [Backport] Fix the issue with customer inline edit when password is expired (by @dmytro-ch)
 - magento/magento2#18400: Admin Login Form > Aliging Label (by @rafaelstz)
 - magento/magento2#18391: Backport 2.2 - Allow keyboard navigation in browser on product detail (by @hostep)
 - magento/magento2#18376: Backport 2.2 - Fix issue 17152 - prevent email being marked as not se (by @sambolek)


Fixed GitHub Issues:
 - magento/magento2#18162: Cannot edit customer using inline edit if password is expired (reported by @JeroenVanLeusden) has been fixed in magento/magento2#18414 by @dmytro-ch in 2.2-develop branch
   Related commits:
     1. 33ba243a068b9e695fdb2520ab463aa3d5911f1a
     2. 2668f2df768d749f2069195715188afbf6d99348

 - magento/magento2#17152: Failure of "Send Order Email Copy" spams customers, every minute, forever. (reported by @lingwooc) has been fixed in magento/magento2#18376 by @sambolek in 2.2-develop branch
   Related commits:
     1. 058c254f0d39a6ab1cad94de42863117404cdfb5

Commit:
merge magento/2.2-develop into magento-pangolin/MQE-1139-2.2

Commit:
ENGCOM-3133: [Backport] Fix wrong return type in StockRegistryInterface API #18427

Commit:
ENGCOM-3128: Backport 2.2 - Don't set a source model on the attribute when it's no… #18390

Commit:
ENGCOM-3126: Fix setup wizard page logo #18403

Commit:
ENGCOM-3114: [Backport] Fixing Snake Case To Camel Case #18425

Commit:
ENGCOM-3111: [BACKPORT] Replace sort callbacks to spaceship operator #18422

Commit:
ENGCOM-3104: Backport 2.2 - Introducing a dedicated cron.log file for logging cron… #18389

Commit:
ENGCOM-3097: [FEATURE] [issue-3283] Added Filter Support for Yes/No (boolean) attr… #17823

Commit:
ENGCOM-3057: Fix table rate failing for zip+4 address #17770 #18166

Commit:
ENGCOM-3050: Fix category tree in cart price rule #17493 #18175

magento-engcom-team committed:
ENGCOM-3137: Admin > Footer > Aligning Proportionally #18401

 - Merge Pull Request magento/magento2#18401 from rafaelstz/magento2:admin-footer-align
 - Merged commits:
   1. 43d3a5fac1607400acffb4a2c12d1211d5e1c943

gelanivishal committed:
Added validation on maximum quantity allowed in shopping cart

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#18427: [Backport] Fix wrong return type in StockRegistryInterface API (by @lewisvoncken)
 - magento/magento2#18390: Backport 2.2 - Don't set a source model on the attribute when it's no (by @hostep)
 - magento/magento2#18403: Fix setup wizard page logo (by @rafaelstz)
 - magento/magento2#18425: [Backport] Fixing Snake Case To Camel Case (by @lewisvoncken)
 - magento/magento2#18422: [BACKPORT] Replace sort callbacks to spaceship operator (by @lewisvoncken)
 - magento/magento2#18389: Backport 2.2 - Introducing a dedicated cron.log file for logging cron (by @hostep)
 - magento/magento2#17823: [FEATURE] [issue-3283] Added Filter Support for Yes/No (boolean) attr (by @lewisvoncken)
 - magento/magento2#18166: Fix table rate failing for zip+4 address #17770 (by @magently)
 - magento/magento2#18175: Fix category tree in cart price rule #17493 (by @magently)


Fixed GitHub Issues:
 - magento/magento2#15085: StockRegistryInterface :: getLowStockItems() returns StockStatusCollection instead of StockItemCollection (reported by @jesse-dev) has been fixed in magento/magento2#18427 by @lewisvoncken in 2.2-develop branch
   Related commits:
     1. adfb130b1bc7074daaeaed466f07a283e2eee8ca

 - magento/magento2#13156: Updating attribute option data through API will set unwanted source_model on the attribute (reported by @koenner01) has been fixed in magento/magento2#18390 by @hostep in 2.2-develop branch
   Related commits:
     1. 3e066db75e495f96d264bf1347b9fec01a44032e

 - magento/magento2#17190: system.log rapidly increasing after Magento CE 2.2.5 update (cron logs) (reported by @mad-develop) has been fixed in magento/magento2#18389 by @hostep in 2.2-develop branch
   Related commits:
     1. a251a8d36ec2c10d175302cac3675e97990bce47

 - magento/magento2#3283: «Yes/No» attributes should be allowed in the Layered Navigation (reported by @dmitry-fedyuk) has been fixed in magento/magento2#17823 by @lewisvoncken in 2.2-develop branch
   Related commits:
     1. 952b333b9908121f17e966078737c21b8620cce6
     2. a2d9a6a5f84d680c23d4f269596baf40ac709395

 - magento/magento2#17770: Table rate fail when using ZIP+4 shipping address (reported by @werfu) has been fixed in magento/magento2#18166 by @magently in 2.2-develop branch
   Related commits:
     1. 6e4a4b2d85e25c58e72914a7904a5dc9aa7e43f5

 - magento/magento2#17493: Catalog Rule & Selected Categories with level > 3 (reported by @SKovbel) has been fixed in magento/magento2#18175 by @magently in 2.2-develop branch
   Related commits:
     1. 9f19592b3a1dc33a8a3a23c694880e5e6c5ca6a3

Commit:
ENGCOM-2997: #4942 and bundle checkbox bug #15905

Commit:
ENGCOM-2629: Added checks to see if the payment is available #15683

Commit:
ENGCOM-2104: Fix type hint of customer-data updateSectionId parameters #16115

ihor-sviziev committed:
magento/magento2#18164 Checkout - Fix "Cannot read property 'code' on undefined" issue

Commit:
merge magento/2.2-develop into magento-pangolin/MQE-1139-2.2

Commit:
Covering the AssignOrderToCustomerObserver by Unit Test

nmalevanec committed:
ENGCOM-2747: Feature australian regions #17516. Remove regions duplication during install.

Commit:
Merge pull request #3193 from magento-pangolin/MQE-1139-2.2

[pangolin] MQE-1139: Cleanup test modules in dev/tests/acceptance in 2.3-develop & 2.2-develop, CE & EE

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

ihor-sviziev committed:
magento/magento2#18164 Checkout - Fix "Cannot read property 'code' on undefined" issue

peterjaap committed:
Fixed typo from filed to field

magento-engcom-team committed:
ENGCOM-3146: Fix type hint of @message declaration as the 'setWidgetParameters' method allows arrays too #18155

 - Merge Pull Request magento/magento2#18155 from avstudnitz/magento2:fix-widget-instance-method-hint
 - Merged commits:
   1. 99f6c1a84637e047996f9d6af3d3e95396c85965

Commit:
Reverted vendor/.htaccess file that was removed in 3434fa1f47aa9abe37b72fc754940736afd4e2e7

Commit:
Fix for MFTF tests build

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#15905: #4942 and bundle checkbox bug (by @JosephMaxwell)
 - magento/magento2#15683: Added checks to see if the payment is available (by @michielgerritsen)
 - magento/magento2#16115: Fix type hint of customer-data updateSectionId parameters (by @Vinai)


Fixed GitHub Issues:
 - magento/magento2#4942: On editing a Bundle product from shopping cart the user defined quantities of the options are overwritten (reported by @densen45) has been fixed in magento/magento2#15905 by @JosephMaxwell in 2.2-develop branch
   Related commits:
     1. c05ac4c243810a7d94517a82680e29cc6371c33b
     2. 076bb60450be1aab2f89b379c68bd7f311819a50
     3. e35a9b1de6bbc961d3bc3acbbd31ca288c893ee5

 - magento/magento2#15652: REST API create order POST /V1/orders (reported by @bibiseb) has been fixed in magento/magento2#15683 by @michielgerritsen in 2.2-develop branch
   Related commits:
     1. f1554d42e7824ea9149d20142a0178f7a7d31f90

magento-engcom-team committed:
ENGCOM-3082: Fix Customer custom attributes lost after save #17968

 - Merge Pull Request magento/magento2#17968 from Thundar/magento2:issue/12479
 - Merged commits:
   1. 7965c9bfd532889bca0752a9ef318dae43feb26c
   2. 44fe03f6f067d4f93819b4353f82e2b8caec6c8f
   3. 6a098a7d21b101d1d1ee058917a15ff6ea4406b4

magento-engcom-team committed:
ENGCOM-3154: [Backport] Checkout - Fix 'Cannot read property 'code' on undefined' issue #18495

 - Merge Pull Request magento/magento2#18495 from ihor-sviziev/magento2:checkout-fix-cannot-read-property-code-of-undefined
 - Merged commits:
   1. e7f70abdd40b9bb1f46c45230412dbda0b421246
   2. 25e3cfe56545df3a4dc502f0fa0d96f63f2e5e88

Commit:
ENGCOM-3146: Fix type hint of @message declaration as the "setWidgetParameters" method allows arrays too #18155

Commit:
ENGCOM-3137: Admin > Footer > Aligning Proportionally #18401

Commit:
ENGCOM-2747: Feature australian regions #17516

magento-engcom-team committed:
ENGCOM-3158: Fix for custom product attribute changing 'backend_type' when 'is_user_defined = 1' and get updated/saved in Admin Backend #18196

 - Merge Pull Request magento/magento2#18196 from bartoszkubicki/magento2:attribute-backend-type-unexpected-change
 - Merged commits:
   1. f6a94595ace124761b9c423f7655b63db48e6545

Commit:
magento-engcom/magento2ce#2240: Fixed issue with directory upgrade script

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#18155: Fix type hint of @message declaration as the "setWidgetParameters" method allows arrays too (by @avstudnitz)
 - magento/magento2#18401: Admin > Footer > Aligning Proportionally (by @rafaelstz)
 - magento/magento2#17516: Feature australian regions (by @maximbaibakov)


Fixed GitHub Issues:
 - magento/magento2#17514: Add Australian regions (reported by @maximbaibakov) has been fixed in magento/magento2#17516 by @maximbaibakov in 2.2-develop branch
   Related commits:
     1. 1b8b0127522bfd41d93514be4741a88e1901d084
     2. b1fcec41c1284fa203a3d0c366a76ada22f1cbec
     3. 020ca367e590f5ac49df197f6002b66af5048544
     4. ebb517b4e4886eb176158bac0412557c5b2b4764

Commit:
ENGCOM-3158: Fix for custom product attribute changing 'backend_type' when 'is_user_defined = 1' and get updated/saved in Admin Backend #18196

Commit:
ENGCOM-3082: Fix Customer custom attributes lost after save #17968

magento-engcom-team committed:
ENGCOM-3171: [Backport] Added validation on maximum quantity allowed in shopping cart #18552

 - Merge Pull Request magento/magento2#18552 from gelanivishal/magento2:2.2-develop-PR-port-18481
 - Merged commits:
   1. db4ac896b7827e73b7948f1744ab40d9c661468f

magento-engcom-team committed:
ENGCOM-3172: [Backport] throw exception InvalidArgumentException during validate scheme #18554

 - Merge Pull Request magento/magento2#18554 from gelanivishal/magento2:2.2-develop-PR-port-18416
 - Merged commits:
   1. 3109909d76b0402a941518bc9306697e4431ef54
   2. 9afdfee251b7ef9098b3ae22e59cf0673c557bb3
   3. 791609c7ec59700694609937deeecd1916fda2b6
   4. beeecafdb28a20b39e06c17554e49bbffa1f7632
   5. 7e38326028a4d2419fa553723c4eef4bd4c90495
   6. a20968db7a29a3b6f26a46979179851121eea336
   7. 2f2cafee687d81d95854b1629ef8eec7c5a4e548
   8. 72b31c952619104e1b176fe9c03074437c8a58b9
   9. a7cdae0e4f80aa8596d4b3a3b958ae2b9938a346
   10. 74c6bfd295569253d2a371bf2f0d11fda931c68c

magento-engcom-team committed:
ENGCOM-3173: Fixed typo from filed to field #18556

 - Merge Pull Request magento/magento2#18556 from gelanivishal/magento2:2.2-develop-PR-port-18505
 - Merged commits:
   1. aedb6c92f2491ab89ebfac1d674a3cbc4d9c6b78

magento-engcom-team committed:
ENGCOM-3176: [Backport] Empty option Label should always be blank even if attribute is required #18564

 - Merge Pull Request magento/magento2#18564 from gelanivishal/magento2:2.2-develop-PR-port-18221
 - Merged commits:
   1. ba9a1b245a2e7cd49dde7000a3ba5b69c136f143

magento-engcom-team committed:
ENGCOM-3178: [Backport] Covering the AssignOrderToCustomerObserver by Unit Test #18559

 - Merge Pull Request magento/magento2#18559 from gelanivishal/magento2:2.2-develop-PR-port-18490
 - Merged commits:
   1. a2b69ed27708cedf575437e9f09e259962828ed1

Commit:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#18196: Fix for custom product attribute changing 'backend_type' when 'is_user_defined = 1' and get updated/saved in Admin Backend (by @bartoszkubicki)
 - magento/magento2#17968: Fix Customer custom attributes lost after save (by @Thundar)


Fixed GitHub Issues:
 - magento/magento2#9219: Custom Product Attribute changes 'backend_type' when 'is_user_defined = 1' and get updated/saved in Admin Backend (reported by @mhauri) has been fixed in magento/magento2#18196 by @bartoszkubicki in 2.2-develop branch
   Related commits:
     1. f6a94595ace124761b9c423f7655b63db48e6545

 - magento/magento2#12479: Saving Customer Model directly causes loss of data (reported by @Danielc3) has been fixed in magento/magento2#17968 by @Thundar in 2.2-develop branch
   Related commits:
     1. 7965c9bfd532889bca0752a9ef318dae43feb26c
     2. 44fe03f6f067d4f93819b4353f82e2b8caec6c8f
     3. 6a098a7d21b101d1d1ee058917a15ff6ea4406b4

Commit:
ENGCOM-3178: [Backport] Covering the AssignOrderToCustomerObserver by Unit Test #18559

Commit:
ENGCOM-3176: [Backport] Empty option Label should always be blank even if attribute is required #18564

Commit:
ENGCOM-3173: Fixed typo from filed to field #18556

Commit:
ENGCOM-3172: [Backport] throw exception InvalidArgumentException during validate scheme #18554

Commit:
ENGCOM-3171: [Backport] Added validation on maximum quantity allowed in shopping cart #18552

Commit:
ENGCOM-3154: [Backport] Checkout - Fix "Cannot read property 'code' on undefined" issue #18495

mahesh-rajawat committed:
fixed issue#18268

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=18583) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=18583) before we can accept your contribution.<br/>**3** out of **9** committers have signed the CLA.<br/><br/>:white_check_mark: sidolov<br/>:white_check_mark: magento-engcom-team<br/>:white_check_mark: dmytro-ch<br/>:x: zakdma<br/>:x: aakimov<br/>:x: StasKozar<br/>:x: mastiuhin-olexandr<br/>:x: yelahin-serhiy<br/>:x: serhii-balko

magento-engcom-team commented:
Hi @maheshWebkul721. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

sidolov commented:
Hi @maheshWebkul721 , looks like something went wrong with this PR, changeset contains 250+ commits. Please, verify your branch and update the PR. 
Thank you!

Commit:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#18559: [Backport] Covering the AssignOrderToCustomerObserver by Unit Test (by @gelanivishal)
 - magento/magento2#18564: [Backport] Empty option Label should always be blank even if attribute is required (by @gelanivishal)
 - magento/magento2#18556: [Backport] Fixed typo from filed to field (by @gelanivishal)
 - magento/magento2#18554: [Backport] throw exception InvalidArgumentException during validate scheme (by @gelanivishal)
 - magento/magento2#18552: [Backport] Added validation on maximum quantity allowed in shopping cart (by @gelanivishal)
 - magento/magento2#18495: [Backport] Checkout - Fix "Cannot read property 'code' on undefined" issue (by @ihor-sviziev)


Fixed GitHub Issues:
 - magento/magento2#14555: Communication's component validator does not propagate exceptions, obscuring the cause of the error (reported by @nikita2206) has been fixed in magento/magento2#18554 by @gelanivishal in 2.2-develop branch
   Related commits:
     1. 3109909d76b0402a941518bc9306697e4431ef54
     2. 9afdfee251b7ef9098b3ae22e59cf0673c557bb3
     3. 791609c7ec59700694609937deeecd1916fda2b6
     4. beeecafdb28a20b39e06c17554e49bbffa1f7632
     5. 7e38326028a4d2419fa553723c4eef4bd4c90495
     6. a20968db7a29a3b6f26a46979179851121eea336
     7. 2f2cafee687d81d95854b1629ef8eec7c5a4e548
     8. 72b31c952619104e1b176fe9c03074437c8a58b9
     9. a7cdae0e4f80aa8596d4b3a3b958ae2b9938a346
     10. 74c6bfd295569253d2a371bf2f0d11fda931c68c

 - magento/magento2#18477: Set maximum Qty Allowed in Shopping Cart is 0 still allow adding to cart (reported by @duongdiep212) has been fixed in magento/magento2#18552 by @gelanivishal in 2.2-develop branch
   Related commits:
     1. db4ac896b7827e73b7948f1744ab40d9c661468f

 - magento/magento2#18164: Checkout - Cannot read property 'code' of undefined (reported by @bruno-serfe) has been fixed in magento/magento2#18495 by @ihor-sviziev in 2.2-develop branch
   Related commits:
     1. e7f70abdd40b9bb1f46c45230412dbda0b421246
     2. 25e3cfe56545df3a4dc502f0fa0d96f63f2e5e88

mahesh-rajawat committed:
Merge remote-tracking branch 'upstream/2.2-develop' into mahesh

mahesh-rajawat committed:
Codacy/PR Quality, updated check condition
